### PR TITLE
fix: Add http agent to axios to work with proxy

### PR DESCRIPTION
**Pull request by david-amores-anz**
Fixes twilio/twilio-cli#223

For the Twilio CLI to work behind an https proxy, we can now set the proxy URL in an env variable called `HTTP_PROXY`. This will set the appropriate `httpAgent` in axios.

### Testing instructions
- Have a locally running forward proxy
- Be on a network behind a https authenticating proxy
- Set environment variable `HTTP_PROXY` to have the proxy's url

On `main`:
- Run any Twilio CLI command
- Make sure the command fails

On this branch:
- Run any Twilio CLI command
- Make sure the command succeeds

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli-core/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ Formatters to take a JSON array and write to the stdout. Current formatters incl
 
 A custom http client for the Twilio helper library to allow us to log API requests as well as modify the User-Agent header.
 
+#### Environment variables
```
Comment (JenniferMah) [inline README.md:33]: Rename `#### Environment variables` to `### Usage with proxy`

File: src/services/cli-http-client.js [hunk 2]
```diff
@@ -17,6 +18,14 @@ class CliRequestClient {
     this.commandName = commandName;
     this.logger = logger;
     this.http = http || require('axios');
+    if (process.env.HTTP_PROXY) {
+      /*
+       * If environment variable HTTP_PROXY is set,
+       * add an appropriate httpsAgent to axios.
+       */
+      this.http.defaults.proxy = false;
```
Comment (JenniferMah) [inline src/services/cli-http-client.js:26]: Why does this need to be set to false?

**Review by JenniferMah — COMMENTED** [review]
Thanks for your PR @david-amores-anz!

File: src/services/cli-http-client.js (hunk 2 above)
Comment (david-amores-anz) [reply src/services/cli-http-client.js:26]: If not explicitly `false`, axios will create a proxy given there is an `http_proxy` env variable set ([line 125](https://github.com/axios/axios/blob/fe52a611efe756328a93709bbf5265756275d70d/lib/adapters/http.js#L125) onwards of their http.js adapter). This proxy will overshadow the one set in the custom http agent and it does not work.

Axios' proxy not working has been reported in a few of their issues such as [here](https://github.com/axios/axios/issues/2072) and [here](https://github.com/axios/axios/issues/925).

File: README.md (hunk 1 above)
Comment (david-amores-anz) [reply README.md]: This is now changed

File: README.md [hunk 3]
```diff
@@ -30,6 +30,9 @@ Formatters to take a JSON array and write to the stdout. Current formatters incl
 
 A custom http client for the Twilio helper library to allow us to log API requests as well as modify the User-Agent header.
 
+### Usage with proxy
+ - `HTTP_PROXY`: If using Twilio CLI behind a proxy, set the URL of the proxy here.
+
```
Comment (JenniferMah) [inline README.md:35]: Can you add some additional information on this being an environment variable that developers will need to set? Maybe something like: 
- `HTTP_PROXY`: If using Twilio CLI behind a proxy, set the URL of the proxy in an environment variable called `HTTP_PROXY`.

Comment (david-amores-anz) [reply README.md:35]: I've updated this now

**Review by eshanholtz — APPROVED** [review]
Thank you @david-amores-anz for the contribution!

**Pull request merged**